File: python/interview-questions/hashes-and-strings/check_permutation.py
def check_permutation(first_text: str, second_text: str) -> bool:
    sorted_first_text: str = "".join(sorted(first_text))
    loop_limit: int = len(second_text) - len(first_text)

    for i in range(0, loop_limit):
        index_step: int = i + len(first_text)
        sub_string: str = second_text[i:index_step]
        sorted_sub_string: str = "".join(sorted(sub_string))

        if sorted_first_text == sorted_sub_string:
            return True

    return False

File: golang/interview-questions/hashes-and-strings/check_permutation.go
package main

import (
	"sort"
)

func SortString(text string) string {
	textToRunes := []rune(text)

	sort.Slice(textToRunes, func(i, j int) bool {
		return textToRunes[i] < textToRunes[j]
	})

	return string(textToRunes)
}

func CheckPermutation(firstText string, secondText string) bool {

	sortedFirstText := SortString(firstText)
	loopLimit := len(secondText) - len(firstText)

	for i := 0; i <= loopLimit; i++ {
		indexStep := i + len(firstText)
		subString := secondText[i:indexStep]
		sortedSubString := SortString(subString)

		if sortedFirstText == sortedSubString {
			return true
		}
	}

	return false
}
